tmt/steps/report/junit.py

```diff
@@ -58,40 +67,230 @@ def duration_to_seconds(duration: Optional[str]) -> Optional[int]:
             f"Malformed duration '{duration}' ({error}).")
 
 
+class ResultWrapper:
+    """
+    Adds possibility to wrap the ``tmt.Result`` and dynamically add more attributes which get
+    available inside the template context.
+    """
+
+    class PropertyDict(TypedDict):
+        """ Defines a property dict, which gets propagated into the final template properties. """
+
+        name: str
+        value: str
+
+    def __init__(self, wrapped: tmt.Result) -> None:
+        self._wrapped = wrapped
+        self._properties: dict[str, str] = {}
+
+    @property
+    def properties(self) -> list[PropertyDict]:
+        return [{'name': k, 'value': v} for k, v in self._properties.items()]
+
+    def set_properties(self, keyval: dict[str, str]) -> None:
```

Comment:
This method does not seem to be called from any place in this PR, is it needed? If it's still necessary, I'd recommend turning it into property setter with `@properties.setter`.

Reply:
This method will be used in the Polarion report plugin:
- https://github.com/teemtee/tmt/pull/3166/files#diff-3f066603b29f42721b98aab30f0d1afac08020970356069510803b03c9b4b234R312

It was originally defined as a `@properties.setter`, but again, I was getting some typing errors that are not showing anymore. I will change it to a property setter in this PR and we can solve possible typing problems in #3166.

Reply:
Sure, I believe we will be able to resolve typing issues.